src/Error/Error.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-|
+Description : The top-level error data types. Specific error types are nested within these.
+-}
+
+module Error.Error where
+
+import Error.TypeError
+import Data.Aeson hiding (Error)
+
+import Text.Parsec.Pos
+
```

Comment:
This data type structure makes it easy to require/ensure that all error messages have some specific info (e.g. source position)

Reply:
Awesome 👍